libs/wingc/src/parser.rs

```diff
@@ -41,6 +41,99 @@ static UNIMPLEMENTED_GRAMMARS: phf::Map<&'static str, &'static str> = phf_map! {
 	"=>" => "see https://github.com/winglang/wing/issues/474",
 };
 
+static RESERVED_WORDS: phf::Set<&'static str> = phf_set! {
+	// JS Reserved Words
+	"abstract",
+	"arguments",
+	"await",
+	"boolean",
+	"break",
+	"byte",
+	"case",
+	"catch",
+	"char",
+	"class",
+	"const",
+	"continue",
+	"debugger",
+	"default",
+	"delete",
+	"do",
+	"double",
+	"else",
+	"enum",
+	"eval",
+	"export",
+	"extends",
+	"false",
+	"final",
+	"finally",
+	"float",
+	"for",
+	"function",
+	"goto",
+	"if",
+	"implements",
+	"import",
+	"in",
+	"instanceof",
+	"int",
+	"interface",
+	"let",
+	"long",
+	"native",
+	"new",
+	"null",
+	"package",
+	"private",
+	"protected",
+	"public",
+	"return",
+	"short",
+	"static",
+	"super",
+	"switch",
+	"synchronized",
+	"this",
+	"throw",
+	"throws",
+	"transient",
+	"true",
+	"try",
+	"typeof",
+	"var",
+	"void",
+	"volatile",
+	"while",
+	"with",
+	"yield",
+
+	// wing-specific reserved words
+	"struct",
+	"as",
+	"nil",
+	"test",
+	"inflight",
+	"preflight",
+	"elif",
+	"init",
+
+	// nodejs globals used in emitted code
+	"require",
+	"module",
+	"process",
+	"Object",
+	"Set",
+
+	// consts declared by the compiler in preflight
+	"$stdlib",
+	"$outdir",
+	"$wing_is_test",
+	"$App",
+	"$AppBase",
+	"$Root",
```

Comment:
I might just skip these since the list might get out of date more quickly (and if a user gets errors by declaring a variable like "$stdlib", it would probably be better to just track that as a compiler bug instead IMO, since there's no technical reason they have to be reserved). Up to you though

Reply:
Makes sense, I guess I'll take them out since there's no chance for them to show up during inflight (which is where it's a much worse bug)